scripts/lib/CIME/XML/env_batch.py

```diff
@@ -195,18 +195,31 @@ def set_job_defaults(self, batch_jobs, pesize=None, walltime=None, force_queue=N
             else:
                 task_count = int(task_count)
 
-            queue = force_queue if force_queue is not None else self.select_best_queue(task_count, job)
-            self.set_value("JOB_QUEUE", queue, subgroup=job)
-
-            walltime = self.get_max_walltime(queue) if walltime is None else walltime
-            if walltime is None:
-                logger.warn("Could not find a queue matching task count %d, falling back to deprecated default walltime parameter"%task_count)
-                #if the user names a queue which is not defined in config_batch.xml and does not set a
-                #walltime, fall back to the max walltime in the default queue
-                if force_queue:
-                    self.get_default_queue()
-                walltime = self._default_walltime
+            if force_queue:
```

Comment:
Yeah, that true. I'm wondering how to support the following case:

A test that uses few procs but takes a long time, the test lives in acme_integration.

1) Due to being in acme_integration, 3 hours is requested
2) Due to using a small number of procs, blues will try to put it in the debug queue

With the changes below, the walltime is automatically reduced and the test will at least have a chance to run.